### DISCUSSION TITLE
Levensfasen - uitwerking 4 - Plan ja/nee als contextinformatie

### DISCUSSION THREAD

**Discussion by pmaria** [Geonovum/disgeo-imsor#33] [General]
Deze modelleringen zijn uitwerkingen  o.b.v. [deze casus](https://github.com/Geonovum/disgeo-imsor/tree/master/issues/15-levensfasen/uitwerkingen) waarbij er voor objecten, wanneer ze gepland worden, of er plannen voor bestaan, dit wordt aangegeven met een context-eigenschap (metadata).

**Comment by pmaria** [comment]
## Alternatief a - Verhardingsplan en hergebruik van objecttype met dezelfde id

In deze uitwerking wordt naast andere context-informatie, zoals geldigheid, een eigenschap `status` toegevoegd bij de `RegistratieGegevens` behorende bij het object. Deze `status` kan de waarde `Gerealiseerd` of `Gepland` hebben, wat het onderscheid maakt tussen gegevens over werkelijk gerealiseerde verhardingen of gegevens over een geplande situatie.

Tijdreizen kun je zien als het bepalen van een context. Het bepalen of een set gegevens gaat over een werkelijk gerealiseerde situatie of een gepande situatie zou je ook een plek kunnen geven in het bepalen van de context waarin gegevens beschouwd moeten worden.

![4-plan-als-context-info](https://github.com/Geonovum/disgeo-imsor/raw/master/issues/15-levensfasen/uitwerkingen/4-plan-als-context-info/4-plan-als-context-info.png)

### Voordelen (wordt bijgewerkt adhv discussie)
- Simpel model
- Kan complexe situatie aan: meerdere plannen per object
- Geen extra schaduw objecttype nodig
- ?

### Nadelen (wordt bijgewerkt adhv discussie)
- Extra contextattribuut = extra complexiteit in gebruik. Je moet je bewust zijn van deze context.
- Statussen moeten obv gegevensregels onderscheiden worden
- ?

**Comment by lvdbrink** [comment]
Simpel model, maar de gebruiker moet wel beter kijken om te zien dat het ene object een plan is, en het andere een bestaand object, en dat ze bij elkaar horen. Net als in uitwerking 2a wordt er gebruik gemaakt van hetzelfde ID, met behulp van de `versie` wordt hier dan onderscheid in aangebracht.

**Comment by RuudKathmann** [comment]
Deze status lijkt mij overbodig, omdat de verschillende statussen die benoemd zijn in het conceptueel model dit onderscheid in voldoende mate helder maken en aanvullend meer informatie geven.

**Reply by pmaria** [reply]
Voor deze uitwerking is de status niet overbodig, maar juist essentieel. De status geeft aan of de gegevens van het object waar het mee verbonden is als gepland moet worden gezien of niet. Dit maakt het mogelijk om eigenlijk twee werkelijkheden naast elkaar te laten bestaan. Een geplande en een gerealiseerde. Hiermee verhef je het wel of niet gepland zijn van gegevens tot onderdeel van je metamodel, op hetzelfde niveau als bijv. de tijdslijn geldigheid.

**Comment by lvdbrink** [comment]
Aanvullingen vanuit overleg 15 april: 

- Er zijn in de groep nogal wat vragen en er is veel uitleg nodig om deze uitwerking te begrijpen. Dat is wel een nadeel van deze uitwerking: moeilijker te begrijpen.
- `status ontwerp ja/nee` is misschien duidelijker, als gegeven bij registratieobject, zo heet het in DSO.
